storage/interface.go

```diff
@@ -52,7 +52,10 @@ type MakeDirer interface {
 // TransactionParams describes a new transaction.
 type TransactionParams struct {
 
-	// RootOverwrite indicates if this transaction will overwrite data at root.
+	// BasePaths indicates the top-level paths where write operations will be performed in this transaction.
+	BasePaths []string
+
+	// RootOverwrite is deprecated. Use BasePaths instead.
 	RootOverwrite bool
```

Comment:
Would code using the TransactionParams before now break because `BasePaths` doesn't equal ""? Or do we handle that backwards-compatibly via `RootOverwrite`? 🤔

Reply:
Good point. We can support backwards-compatibly via `RootOverwrite` for now and then remove it in a future release. Updated the code.